### PR TITLE
Refactor/dynamite runtime

### PR DESCRIPTION
**Pull request by provokateurin**

### CODE REVIEW

File: packages/nextcloud/lib/src/nextcloud.openapi.dart [hunk 1]
```diff
@@ -10,68 +10,11 @@ import 'package:built_value/serializer.dart';
...
-  NextcloudResponse(
-    super.data,
-    super.headers,
-  );
-
-  @override
-  String toString() => 'NextcloudResponse(data: $data, headers: $headers)';
-}
-
-class NextcloudApiException extends DynamiteApiException {
```
Comment (Leptopoda) [inline packages/nextcloud/lib/src/nextcloud.openapi.dart:29]: I'd still like having a separate exception.

An app could also include a nextcloud unrelated openapi schema and this makes it impossible to distinguish them.

File: packages/nextcloud/lib/src/nextcloud.openapi.dart [hunk 2]
```diff
@@ -10,68 +10,11 @@ import 'package:built_value/serializer.dart';
...
 part 'nextcloud.openapi.g.dart';
 
-class NextcloudResponse<T, U> extends DynamiteResponse<T, U> {
-  NextcloudResponse(
-    super.data,
-    super.headers,
-  );
-
-  @override
-  String toString() => 'NextcloudResponse(data: $data, headers: $headers)';
```
Comment (Leptopoda) [inline packages/nextcloud/lib/src/nextcloud.openapi.dart:26]: same as below.
The toString cuttently contains the prefix wich could make debugging easier.

**Review by Leptopoda — COMMENTED** [review]
Having the Auth in dynamite_runtime makes sense ut we should still have custom types to better distinguish/debug the case where we have multiple separate generated apis.

File: packages/nextcloud/lib/src/nextcloud.openapi.dart (hunk 1 above)
Comment (provokateurin) [reply packages/nextcloud/lib/src/nextcloud.openapi.dart:29]: Why would it? We have stacktraces, right?

Comment (Leptopoda) [reply packages/nextcloud/lib/src/nextcloud.openapi.dart:29]: I was acrually thinking of trying to handle the error.
Imagine hypothetical case where a dev might want to generate two separate specs (dynatime should be general purpose) and the dev might want to handle the errors differently (in a try catch or the UI).

For this they'd need to know what they are dealing with.

This is just dynamite stuff and we'd throw the spec specific one (NewsError, NotesError) but neon could just handle the parent (DynamiteError).

Comment (provokateurin) [reply packages/nextcloud/lib/src/nextcloud.openapi.dart:29]: Makes sense, although I don't know if anyone will ever have this use-case. As you said for neon we should just handle the generic DynamiteException.

**Review by Leptopoda — APPROVED** [review]

**Pull request merged**